### PR TITLE
added new api to get all issued esdts

### PR DESCRIPTION
**Pull request by sasurobert**
added new api to get all issued esdts

### CODE REVIEW

File: node/node.go [hunk 1]
```diff
@@ -444,6 +445,48 @@ func (n *Node) GetUsername(address string) (string, error) {
 	return string(username), nil
 }
 
+// GetAllIssuedESDTs returns all the issued esdt tokens, works if not is on metachain
+func (n *Node) GetAllIssuedESDTs() ([]string, error) {
+	if n.shardCoordinator.SelfId() != core.MetachainShardId {
+		return nil, ErrAccountNotFound
+	}
+
+	account, err := n.accounts.GetExistingAccount(vm.ESDTSCAddress)
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply node/node.go:454]: yes. will do.

File: node/node.go [hunk 2]
```diff
@@ -444,6 +445,48 @@ func (n *Node) GetUsername(address string) (string, error) {
 	return string(username), nil
 }
 
+// GetAllIssuedESDTs returns all the issued esdt tokens, works if not is on metachain
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply node/node.go:448]: done

File: facade/nodeFacade.go [hunk 3]
```diff
@@ -287,6 +287,11 @@ func (nf *nodeFacade) GetAllESDTTokens(address string) (map[string]*esdt.ESDigit
 	return nf.node.GetAllESDTTokens(address)
 }
 
+// GetAllIssuedESDTs returns all the issued esdts for a given address
```
Comment (SebastianMarian) [inline facade/nodeFacade.go:290]: It is not for a given address, refactor the comment

File: node/node.go [hunk 4]
```diff
@@ -444,6 +445,44 @@ func (n *Node) GetUsername(address string) (string, error) {
...
+
+	ctx := context.Background()
+	chLeaves, err := userAccount.DataTrie().GetAllLeavesOnChannel(rootHash, ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	for leaf := range chLeaves {
+		key := string(leaf.Key())
+		if strings.Contains(key, "-") {
```
Comment (SebastianMarian) [inline node/node.go:478]: So "-" should be reserved only for esdt as a symbol used in the key

File: cmd/node/config/api.toml [hunk 5]
```diff
@@ -71,7 +71,10 @@
 
         # /network/config will return metrics related to current configuration of the network (number of shards,
         # consensus group size and so on)
-        { Name = "/config", Open = true }
+        { Name = "/config", Open = true },
+
+        # /network/esdts will return all the issued esdts on the protocol
+        { Name = "/esdts", Open = true },
```
Comment (bogdan-rosianu) [inline cmd/node/config/api.toml:77]: remove trailing `,`

File: facade/nodeFacade.go (hunk 3 above)
Comment (bogdan-rosianu) [inline facade/nodeFacade.go:290]: `// GetAllIssuedESDTs returns all the issued esdts from the esdt system smart contract`

File: node/node.go (hunk 4 above)
Comment (bogdan-rosianu) [inline node/node.go:478]: is this enough? we have to make sure no other key would contain `-`

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: cmd/node/config/api.toml (hunk 5 above)
Comment (sasurobert) [reply cmd/node/config/api.toml:77]: done

File: facade/nodeFacade.go (hunk 3 above)
Comment (sasurobert) [reply facade/nodeFacade.go:290]: done

File: node/node.go (hunk 4 above)
Comment (sasurobert) [reply node/node.go:478]: this is enough

File: facade/nodeFacade.go (hunk 3 above)
Comment (sasurobert) [reply facade/nodeFacade.go:290]: done

File: node/node.go (hunk 4 above)
Comment (sasurobert) [reply node/node.go:478]: this is ok.

**Review by bogdan-rosianu — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Pull request merged**